Submission of 1749_Cloud-Native_Standards_Checklist.md

joshhalley opened the pull request:
As agreed with Riaan and Ricardo please find the PR for the Agentic standards checklist. 

Add Issue as deliverable: https://github.com/cncf/toc/issues/1749#issuecomment-2998733687

linsun commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,362 @@
+
+General best practices include: **Security**, which covers minimizing attack surface and safeguarding container integrity; **Observability**, which focuses on collecting actionable metrics, logs, and traces to understand system behavior; and **Availability & Fault Tolerance**, which outlines strategies for maintaining service continuity and resilience under failure conditions.
+
+**Security**
```
could we highlight what is unique to AI agents so it is clear to the reader?  It could even be helpful to explain why these are unique.

joshhalley replied:
Discussed to item with Nina - its a good call out, as we agree this could be confusing. Essentially we aim at having the general section setting the stage in terms of standard best practices, as a precursor to the agent specific sections. We will update the section to make a clearer transition between sections.

linsun commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,362 @@
+
+# Control and Communication 
+
+Microservice architectures have long followed the principles and practices of information hiding, minimal endpoint exposure for only the requisite and needed services, and clear contract-based communications. These same principles should be followed in the context of Agentic architectures. As multi-agent systems grow, the intricacy of ensuring effective coordination and communication rises signif ...
```
isn't ACP being merged to A2A?

linsun replied:
worth highlight secure communication become way more important as unpredictable nature of ai agents.

joshhalley replied:
Updated in last commit

linsun commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,362 @@
+- [ ] **Protocols today (MCP,  A2A, etc…)**   
+      While there is a broad array of competing technologies in the industry today, numerous protocols have gained a level of interest for preliminary evaluation in the field in the domain of Agentic AI. Each of these protocols aims at addressing a specific problem space within the domain, ranging from tool exposure to inter-framework communications or agent discovery, trust and identification se ...
+      - [ ] MCP from Anthropic: Provides a key focus on “tool exposure” providing a means to define an “MCP Server” which is responsible for providing tool access to an “MCP Client”, in many scenarios the use of MCP can support both single and multi-agent frameworks in achieving standardized access to specific tooling, without the need to define and program the logic from scratch. Considerations  ...
+      - [ ] A2A from Google: Offers a means for agents to communicate with one another directly, analogous to peer-to-peer communications. Providing an optimal means for agents which may even exist in disparate domains or that utilize disparate frameworks, with the means to establish communications with one another. A2A utilizes JSON-RPC 2.0 over HTTPS and Server-Sent Events (SSE).   
```
Aren't MCP also use json-rpc and supports SSE?  Felt odd it is only mentioned in A2A

joshhalley replied:
Correct - will update

joshhalley replied:
Updated in last commit

linsun reviewed (commented):
left a few comments, will finish review in the next few days

linsun commented:
Hi @joshhalley - Could you also clarify who are the expected readers of this paper? Thanks!

joshhalley commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,362 @@
+- [ ] Utilize PodDisruptionBudgets to enforce minimum pod availability during voluntary disruptions like upgrades or node drains.  
+- [ ] Use Pod Anti‑Affinity or Topology Spread Constraints to distribute pod replicas across nodes or zones, minimizing the impact of node or zone-level failures when possible.  
+- [ ] Use Horizontal Pod Autoscaler (HPA) to scale workloads dynamically using CPU, Memory, or custom metrics such as request volume.  
+- [ ] Inference extensions provided via the Gateway API ensure that path-based rules are to be applied with a focus on inference served AI Models. This capability supports more dynamic deployment scenarios used by agents.
```
We will call this out more specifically - being a more general best practice, and not necessarily applicable to all scenarios such as the ones which you rightly noted.

joshhalley commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,362 @@
+      While there is a broad array of competing technologies in the industry today, numerous protocols have gained a level of interest for preliminary evaluation in the field in the domain of Agentic AI. Each of these protocols aims at addressing a specific problem space within the domain, ranging from tool exposure to inter-framework communications or agent discovery, trust and identification se ...
+      - [ ] MCP from Anthropic: Provides a key focus on “tool exposure” providing a means to define an “MCP Server” which is responsible for providing tool access to an “MCP Client”, in many scenarios the use of MCP can support both single and multi-agent frameworks in achieving standardized access to specific tooling, without the need to define and program the logic from scratch. Considerations  ...
+      - [ ] A2A from Google: Offers a means for agents to communicate with one another directly, analogous to peer-to-peer communications. Providing an optimal means for agents which may even exist in disparate domains or that utilize disparate frameworks, with the means to establish communications with one another. A2A utilizes JSON-RPC 2.0 over HTTPS and Server-Sent Events (SSE).   
+      - [ ] Identity: The Agntcy identity framework (recently donated to the Linux Foundation) takes the approach of allowing varied identity providers to be used in a BYOI (Bring Your Own Identity) construct. What is notably different with this approach for identity is that it also supports the Web3 DID based standard, which supports distributed identity concepts. A novel approach towards the de ...
```
We will introduce it in the communications section for completeness, mentioning its usefulness around workload security and federation, indicating that it is covered in more detail in the security section.

pacoxu commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,362 @@
+# Kubernetes Agentic Standards
```
Cloud Native Agentic?

joshhalley replied:
Agree that the naming needs improvement to avoid confusion - but I would expect that this would take place during the proper publishing of the document. The current markdown version is just a precursor to a final version.

pacoxu commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,362 @@
+
+Agentic systems provide the means to perform multi-hop reasoning, and subsequent action calling based upon signals to augment and provide dynamism to conventional programming sequences.  
+
+This paper explores six key areas which are associated with components that are in need of standardization to ensure interoperability, security, and observability are applied from the beginning. The focus of this document is not on how a particular agentic protocol itself is programmed, which language is used or its speed or efficiency of execution, but rather an agnostic view of what should be c ...
```
What's the six key areas?

joshhalley replied:
Good catch Paco - we consolidated after the document began, we have 4 key areas now, I updated the document.

Control Communications
Observability
Governance
Security

pacoxu commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,362 @@
+
+***A microservice based application which can consist of one or more containers. Performing standalone or collaborative (distributed) function through the interaction with at least one entity (container) or quorum, accessing AI or Machine Learning capabilities. To perform tasks based upon the reasoning and execution of actions within event based systems triggered or reacted upon via signals. Key  ...
+
+Overview
```
Is this a header?

joshhalley replied:
Typo - it should just be bold - as its a starting definition

pacoxu commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,362 @@
+- [ ] Use a standard observability stack (MELT). Consolidating metrics, logs, and traces to ensure viable explainability and achievable debugability of the system.   
+- [ ] Incorporate network observability by collecting flow logs for security, performance monitoring, and troubleshooting.  
+- [ ] Monitor resource-specific metrics  
+      - [ ] Disk Usage: Monitor disk usage on nodes and persistent volumes to prevent outages caused by storage exhaustion.  
```
The preview shows no checkbox with `      `. Similarly, below and other blocks below. 
```suggestion
    - [ ] Disk Usage: Monitor disk usage on nodes and persistent volumes to prevent outages caused by storage exhaustion.  
```

joshhalley commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,362 @@
+- [ ] Continuously monitor and update base images of the container to include the latest security patches and avoid known vulnerabilities. Use distroless images where possible.  
+- [ ] Log and monitor container activity. Monitor runtime behavior, resource usage, filesystem access, network activity, and system calls to detect anomalies or security incidents early.
+
+**Observability**
```
Update added to last commit

joshhalley commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,362 @@
+      - [ ] Given the unpredictable nature of generative AI, defining schemas using JSON Schema, Protobuf, or OpenAPI to validate payloads during tool calls and external service invocations can increase system predictability and avoid cascading failures. Data constraints can be enforced to avoid malformed input, malicious content injection, or drift caused by inconsistent formats.
+
+- [ ] **Protocols today (MCP,  A2A, etc…)**   
+      While there is a broad array of competing technologies in the industry today, numerous protocols have gained a level of interest for preliminary evaluation in the field in the domain of Agentic AI. Each of these protocols aims at addressing a specific problem space within the domain, ranging from tool exposure to inter-framework communications or agent discovery, trust and identification se ...
```
Reworded in last commit

hzxuzhonghu commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,369 @@
+- [ ] Use Horizontal Pod Autoscaler (HPA) to scale workloads dynamically using CPU, Memory, or custom metrics such as request volume.  
+- [ ] Inference extensions provided via the Gateway API ensure that path-based rules are to be applied with a focus on inference served AI Models. This capability supports more dynamic deployment scenarios used by agents.
+
+NOTE: The above items are general in nature, and while applicable to smart load-balancing to inference models, does not pertain to more comprehensive MCP / Agent to Agent or LLM tooling. 
```
There should be a separate networking section

hzxuzhonghu commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,369 @@
+
+- [ ] **Protocols today (MCP,  A2A, etc…)**   
+      While there is a broad array of complimentary technologies in the industry today, numerous protocols have gained a level of interest for preliminary evaluation in the field in the domain of Agentic AI. Each of these protocols aims at addressing a specific problem space within the domain, ranging from tool exposure to inter-framework communications or agent discovery, trust and identificatio ...
+      - [ ] MCP from Anthropic: Provides a key focus on “tool exposure” providing a means to define an “MCP Server” which is responsible for providing tool access to an “MCP Client”, in many scenarios the use of MCP can support both single and multi-agent frameworks in achieving standardized access to specific tooling, without the need to define and program the logic from scratch. Considerations  ...
```
Server-Sent Events  is an older version protocol

ChaoyiHuang commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,369 @@
+
+## General best practices for containers:
+
+The containerization principles outlined herein include emerging definitions of agentic microservices (e.g., autonomous, signal-driven, reasoning-capable container systems), the recommendations in this general section are not specific to agentic use cases and apply to any containerized environment.
```
> agentic microservices

some thoughts here:

the document assumed that AI agents in multi-agent system will employ microservices based architecture.

but AI agents can also run as serverless application, not mandatory have to run as microservices, and serverless is also one of practices in cloud native definition: https://github.com/cncf/toc/blob/main/DEFINITION.md

microservices mostly means long-running services. For AI agent, the large portion of time is in the phase of inference executed in XPU, from handreds milliseconds to seconds, and one task executed by AI agent may include many times inference executions. compared to the time spent in the inference phases, the time to start AI agent(start container with image pulling or not, or just instantiate AI agent from remote database) will become negligible especially when the image is mostly cached in the host. Thus, AI agent running in serverless mode is quite common.

whether the serveless based AI agent system also be covered in the checklist

joshhalley commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,369 @@
+      - [ ] SPIFFE/SPIRE: Focused towards workload security, this technology stack provides a means to secure cryptographic identity of worksloads through the use of SVIDs in the form of JWT or X.509 documents. The solution allows for workload attestation and federation. (This technology is covered more comprehensively in the security section) 
+
+- [ ] **Message and Communication Design Considerations with REST, GRPC & Kafka**   
+      - [ ] To accommodate the data, the usage of known event-driven bus architectures such as Kafka and Flink should be considered. Event buses are especially useful when asynchronous communication is desired (e.g. long-running tasks) or when building an event-driven architecture (e.g. emitting telemetry, decision logs, or coordination signals). Kafka guarantees high reliability and delivery gua ...
```
We have updated the section to map more specifically to agent behaviour.

ChaoyiHuang commented:
sorry to raise comment so later, is it possible to organize these checklist items into more catchy and trendy structure, like following, checklist items could be put under each principle/factor accordingly

[15 Principles for designing and deploying scalable applications on Kubernetes](https://www.cncf.io/blog/2022/02/17/principles-for-designing-and-deploying-scalable-applications-on-kubernetes/)

1. [Never Single Pod](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#single-pod)
2. [Stateful vs. Stateless](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#stateful-stateless)
3. [Secrets vs. Non-secrets](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#secret-non-secret)
4. [Autoscaling](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#auto-scaling)
5. [Lifecycle Management](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#lifecycle-mgmt)
6. [Probes](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#probes)
7. [Fail Hard, Fast, Loud](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#fail-hard)
8. [Prepare Application](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#prep-app)
9. [Resource Requests, limits](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#pod-resources)
10. [Reservation & prioritization](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#capacity)
11. [Scheduling requirements](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#spread)
12. [Pod Disruption Budget](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#disruption-budget)
13. [Strategies > stop the world](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#stop-the-world)
14. [Restrict permissions](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#pod-permissions)
15. [Limit attack surface](https://elastisys.com/designing-and-deploying-scalable-applications-on-kubernetes/#pod-security)

[12-Factor Agents](https://github.com/humanlayer/12-factor-agents)

1. [Factor 1: Natural Language to Tool Calls](https://github.com/humanlayer/12-factor-agents/blob/main/content/factor-01-natural-language-to-tool-calls.md)
2. [Factor 2: Own your prompts](https://github.com/humanlayer/12-factor-agents/blob/main/content/factor-02-own-your-prompts.md)
3. [Factor 3: Own your context window](https://github.com/humanlayer/12-factor-agents/blob/main/content/factor-03-own-your-context-window.md)
4. [Factor 4: Tools are just structured outputs](https://github.com/humanlayer/12-factor-agents/blob/main/content/factor-04-tools-are-structured-outputs.md)
5. [Factor 5: Unify execution state and business state](https://github.com/humanlayer/12-factor-agents/blob/main/content/factor-05-unify-execution-state.md)
6. [Factor 6: Launch/Pause/Resume with simple APIs](https://github.com/humanlayer/12-factor-agents/blob/main/content/factor-06-launch-pause-resume.md)
7. [Factor 7: Contact humans with tool calls](https://github.com/humanlayer/12-factor-agents/blob/main/content/factor-07-contact-humans-with-tools.md)
8. [Factor 8: Own your control flow](https://github.com/humanlayer/12-factor-agents/blob/main/content/factor-08-own-your-control-flow.md)
9. [Factor 9: Compact Errors into Context Window](https://github.com/humanlayer/12-factor-agents/blob/main/content/factor-09-compact-errors.md)
10. [Factor 10: Small, Focused Agents](https://github.com/humanlayer/12-factor-agents/blob/main/content/factor-10-small-focused-agents.md)
11. [Factor 11: Trigger from anywhere, meet users where they are](https://github.com/humanlayer/12-factor-agents/blob/main/content/factor-11-trigger-from-anywhere.md)
12. [Factor 12: Make your agent a stateless reducer](https://github.com/humanlayer/12-factor-agents/blob/main/content/factor-12-stateless-reducer.md)

ChaoyiHuang commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,374 @@
+      
+
+- [ ] **Tools & Services**  
+      - [ ] A common approach is pursued to access tools (MCP, A2A, ACP, etc…). Unnecessary entropy **can lead to complexity** in the ability to operate and monitor the solution effectively. While the correct tool should be considered for the task required, careful consideration should be given to whether variance in the system/solution is necessary.   
```
how about add skills here. currently some tools access will be done by scripts / cli under skill. for example, the popular moltbot(clawdbot) skills, and cluade skills https://github.com/anthropics/skills/tree/main/skills

joshhalley replied:
@ChaoyiHuang We added a reference to the skills in the footnotes for the section, further explanation below in the response to Lin.

joshhalley commented:
> Hi @joshhalley and team, I reviewed the google doc link (assuming it is same as the github PR), I think it LGTM mostly. I noted a few things I'd like to see it corrected/updated:
> 
> MELT - please spell it out MCP from Anthropic — can we clarify it is in AAIF now?
> 
> The Agntcy identity framework - do we need to mention it? I feel it doesn’t have momentum like MCP or A2A yet and I prefer to mention it when project is being adopted by a few adopters. Note there is also A2P and agent skills that are not mentioned which may well have more momentums.
> 
> MCP Project: https://www.anthropic.com/news/model-context-protocol - can we replace with a vendor neutral link, modelcontextprotocol.io?
> 
> https://github.com/plurai-ai/intellagent why would it be mentioned? I’d prefer not mentioning it as it is not in a foundation. And it doesn’t look very popular either.
> 
> https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/gpu-operator-confidential-containers.html - CoCo is a cncf project, can we replace this with a vendor neutral link?

Hi @linsun, thanks a lot for taking time to evaluate this submission, and your well thought out comments, really appreciate them. 

We have gone ahead and added the following changes in the last commit: 

- MELT: updated to be fully spelt out
- MCP: updated, and spelt out, also added the AAIF reference
- Identity: Adjusted it to indicate that its emergent, this protocol is unique based on its capability in IOT / Airgap and distributed deployment environments given its support for DID concepts, during initial authoring the team discussed this, which led to its inclusion.  For A2P, good call out, this was just released at the time the paper was first pushed, I have gone ahead and added it - also has some interesting aspects when it comes to x402 usage.
- Agent Skills - I chatted with Nina on this yesterday, essentially an Agent skill is nothing more than a for loop introducing a function call for context expansion. While this is being seen as something which is lowering the bar for users who lack programming skills to design agents, the usage is more limited to direct consumer (coding agent) flows. I added a link to it in the footnotes section, but don't feel that it rivals well structured programatic routines to execute more surgical tool/function calling at this point in time. 
-  This was added by Vincent to ground it as example of the concept he was describing (I removed it)
- I added the CoCo link for the CNCF project - I did not remove NVIDIA link, in fact I added a further link here to AMD here too. The reason being, is that we are not calling out confidential computing in general, rather the focus towards GPU usage to protect model security at runtime, this functionality is extremely hardware dependent. On the market today, there are very few GPUs which actually support this capability (NVIDIA H100/AMD Instinct) hence I found their inclusion appropriate to compliment the topic and support readers who require a broader context on how they can move forward in this direction for this security functionality

linsun reviewed (approved):
LGTM

Thanks @joshhalley and team!

Please fix the bot :)

linsun commented:
Hi @joshhalley @raravena80 @npolshakova and team - we had a discussion in our TOC meeting on this.  A few additional feedback before publishing this:

- why is this not called "Cloud Native" vs the current Kubernetes agentic standards? A similar suggestion earlier: https://github.com/cncf/toc/pull/1903/changes#r2431262832
- we would like a separate file that indicates how updates can be made to the paper going forward, what is the process for that.
- @dims from the TOC plans to review and provide feedback on this.

The TOC is also working on guidance for tech paper/whitepaper, and @angellk is leading on this effort. This may be ready in a few weeks but if you have addressed the above feedback (pending @dims 's) before the guideline is out, we'd be happy to move forward with publishing this.

dims commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,381 @@
+
+This paper aligns to the definition of Containerization of Agentic AI as: 
+
+**A microservice based application which can consist of one or more containers based upon defined system architectual requirements. Performing standalone or collaborative (distributed) function through the interaction with at least one entity (container) or quorum, accessing AI or Machine Learning capabilities. To perform tasks based upon the reasoning and execution of actions within event based  ...
```
Since this is a definition, i am being a bit picky here to make it more readable, straight forward and from a single author. Please feel free to incorporate or ignore the entire suggestion or parts of it.

```suggestion
**A microservice-based application, composed of one or more containers as dictated by system architecture, that operates either independently or as part of a distributed collaboration—interacting with at least one other entity (container) or achieving quorum-based consensus. It leverages AI or Machine Learning capabilities to reason and execute actions within event-driven systems, where behavior is triggered or modulated by signals. Its defining attributes: full autonomy in executing system or user tasks, coupled with the ability to plan, orchestrate, and govern the continuation or completion of its own execution.**
```

joshhalley commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,381 @@
+
+This section defines the security considerations involved in building agentic systems. Three primary goals should guide the design: authentication, authorization, and trust. Agents and their components must be able to securely authenticate and should only be granted the minimum permissions necessary to function. Trust boundaries must be clearly established to prevent privilege escalation, data le ...
+
+Keeping these goals in mind user access, agent identity, tenancy, and data access must be deliberately designed and enforced. Each agent should have a unique, verifiable identity to support traceability, accountability, and secure communication across system boundaries. Strong tenancy isolation is critical, especially in multi-tenant environments, to prevent cross-agent interference and ensure th ...
```
There is a complete section for agent tenancy which points to RBAC and other concepts in the context of agents. What is not included, is topics around confidential computing would would align more to the compute and physical deployment aspects of ring-fencing deployments. For more elaborate explanations around model security and tenancy, there is a pending security whitepaper under review by the TOC which we consider as a the definitive source of truth.

ChaoyiHuang commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,381 @@
+# Cloud Native Agentic Standards
```
A 'Standard' implies a prescriptive and mandatory requirement where compliance is strictly enforced. Since the items in this checklist are primarily best practices, perhaps 'Guidelines' would be a more accurate term? （considering that the AI Agent field is still rapidly evolving and has yet to reach a stable implementation paradigm）

linsun commented:
Hi @dims could you approve it in github? That will trigger the merge without bypass rules :)

mrbobbytables commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,362 @@
+- [ ] Use Horizontal Pod Autoscaler (HPA) to scale workloads dynamically using CPU, Memory, or custom metrics such as request volume.  
+- [ ] Inference extensions provided via the Gateway API ensure that path-based rules are to be applied with a focus on inference served AI Models. This capability supports more dynamic deployment scenarios used by agents.
+
+![][image1]
```
Yeah - I think this is an issue with different markdown renderers. Right now in its current form the images will not be viewable on github. It might work if done directly as html instead of markdown link.

kerthcet commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,381 @@
+
+This paper aligns to the definition of Containerization of Agentic AI as: 
+
+**A microservice-based application, composed of one or more containers as dictated by system architecture, that operates either independently or as part of a distributed collaboration—interacting with at least one other entity (container) or achieving quorum-based consensus. It leverages AI or Machine Learning capabilities to reason and execute actions within event-driven systems, where behavior  ...
```
> A microservice-based application

This reminds me that the agentic system is based on microservice, which is usually request-response pattern, scaling based on requests etc., however, agentic system is quite different as mentioned after, will this leads to misunderstanding of the system. And microservice is by default described as a distribution system, but agent maybe not? Or we're here only talking about distributed agentic systems since we're running on Kubernetes.

kerthcet reviewed (commented):
Left some comments for discussion, will continue the work later. Thanks!

kerthcet commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,381 @@
+
+- [ ] **Message and Communication Design Considerations with REST, GRPC & Kafka**   
+      - [ ] To accommodate the data, the usage of known event-driven bus architectures such as Kafka and Flink should be considered. Event buses are especially useful when asynchronous communication is desired (e.g. long-running tasks) or when building an event-driven architecture (e.g. emitting telemetry, decision logs, or coordination signals). Kafka guarantees high reliability and delivery gua ...
+      - [ ] gRPC, the rate of data at input for streaming data should be considered carefully to assess whether a ELT or ETL approach needs to be taken. Other considerations around the impact that the amount of streaming data may have on agent token limits when interfacing with an models are worthy of evaluation.   
```
Has the same problem with display.

kerthcet commented on initiatives/1749_Cloud-Native_Standards_Checklist.md:
```diff
@@ -0,0 +1,381 @@
+            - [ ] Environmental Impact (CO2, Carbon Credits)  
+            - [ ] Human Cost (Oversight / Setup Time)   
+            - [ ] Mandatory storage of data (backups, audit trails)   
+      - [ ] **Targeted Use Case Evaluation**  
```
Why highlight Targeted Use Case Evaluation here, or it should be parallel to the `Assessment Criterion`.
